### PR TITLE
Refactor driver scripts for examples.

### PR DESCRIPTION
**Pull request by coryan**
The drivers to run the examples during the CI build were masking some
failures. Refactor them to make this less likely and (I think) easier to maintain.

I am planning to reuse this function (after some refactoring) in the storage tests too,
because there is too much magic in those, e.g.:

https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/d00b5214b0468890eb0d374d1fd3bc5fdae95069/google/cloud/storage/examples/run_examples_utils.sh#L89

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/993)

### CODE REVIEW

File: google/cloud/bigtable/examples/run_examples_utils.sh [hunk 1]
```diff
@@ -72,97 +131,48 @@ function run_all_instance_admin_examples {
   shift 2
 
   local setenv="env"
```
Comment (mbrukman) [inline google/cloud/bigtable/examples/run_examples_utils.sh]: This line no longer necessary?

Comment (coryan) [reply google/cloud/bigtable/examples/run_examples_utils.sh]: Yes. After we stopped running the instance admin integration tests in production this is no longer needed.  All the tests in a group (admin, data, instance admin) use the same environment variables.

Comment (coryan) [reply google/cloud/bigtable/examples/run_examples_utils.sh]: Doh! You meant that I could remove that line... yes, I can.  Will do in the next PR.

Comment (mbrukman) [reply google/cloud/bigtable/examples/run_examples_utils.sh]: Sorry for the subtlety; yes, I did mean that if you don't need it, feel free to remove it. 😃

**Pull request merged**